e2e/specs/translation-builder-workflows.spec.ts

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable testing-library/prefer-screen-queries */
+
+import { test } from '../core';
+import { expect } from '@playwright/test';
+import { deleteForm, createForm, createValueReference, addFormResources } from '../commands/form-operations';
+import { FormBuilderPage } from '../pages';
+
+let formUuid = '';
+
+const formDetails = {
+  name: 'Covid-19 Screening',
+  description: 'A test form for recording COVID-19 screening information',
+  version: '1.0',
+  published: true,
+};
+
+test.describe('Translation Builder Workflows', () => {
+  test.beforeEach(async ({ api }) => {
+    const form = await createForm(api, false, formDetails);
+    formUuid = form.uuid;
+
+    const valueReference = await createValueReference(api);
+    await addFormResources(api, valueReference, formUuid);
+  });
+
+  test.afterEach(async ({ api }) => {
+    if (formUuid) {
+      await deleteForm(api, formUuid);
+    }
+  });
+
+  test('Workflow 1: Basic Translation Builder Display', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I visit the form builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+    });
+
+    await test.step('And I search for the form and open it for editing', async () => {
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+    });
+
+    await test.step('And I open the Translation builder tab', async () => {
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('Then I should see the translation builder components (Download/Upload buttons, Translation filter tabs: All/Translated/Untranslated, and the search input)', async () => {
+      const translationPanel = formBuilderPage.page
+        .getByRole('tabpanel')
+        .filter({ has: formBuilderPage.page.getByRole('button', { name: /upload translation/i }) });
+
+      await expect(translationPanel).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /download translation/i })).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /upload translation/i })).toBeVisible();
+      const tabList = translationPanel.getByRole('tablist', { name: 'Translation filter' });
+      await expect(tabList).toBeVisible();
+
+      const tabs = tabList.getByRole('tab');
+      await expect(tabs).toHaveCount(3);
+      await expect(tabs.nth(0)).toContainText(/all/i);
+      await expect(tabs.nth(1)).toContainText(/translated/i);
+      await expect(tabs.nth(2)).toContainText(/untranslated/i);
+      await expect(translationPanel.getByPlaceholder(/search translation keys/i)).toBeVisible();
+    });
+  });
+
+  test('Workflow 2: Language Selection and Switching', async ({ page }) => {
```

Comment:
Please see the other e2e tests, the canonical way is to describe the workflow in natural language:
```suggestion
  test('Add and view translation for a specific language ', async ({ page }) => {
```
This entire test, you just check if certain elements in the translation builder is present. That shouldn't be the case, each workflow should have a user objective. For example, one workflow would be if you can edit the translations for a certain language. In that case the steps would be:
1. When I visit the form builder
2. And I search for the form I need to edit
3. And I click the `Edit` button on the form I need to edit
4. And then I click the `Translation Builder` tab
5. And I change the language to French
6. And I click on the `Edit string` button on the string I want to add a translation for
7. And then I add the translation
8. And then I click on save to save the translation for the string
9. And then I click on `Render changes`
10. Then I should see the form in French with the added translation

You should update all the other workflows to be the same, I highly suggest you work on one workflow at a time since you are a bit confused.